library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java

```diff
@@ -518,8 +517,9 @@ public <T> T execute(final OperationChain<T> opChain) {
         try {
             LOGGER.info("GafferPop operation chain called: " + opChain.toString());
             return graph.execute(opChain, user);
-        } catch (final OperationException e) {
-            throw new RuntimeException(e);
+        } catch (final Exception e) {
```

Comment:
I think we should be catching all the specific exceptions and then have the exception catch the others.

Reply:
This is the usual practice for Java but not how things have been done in Gaffer. If we want to change to using this approach it would be better to raise as a separate issue.

There's a need to investigate the kinds of exceptions thrown by Operations, ensure they are all children of `OperationException` and enforce this. This is why `Exception` has been directly caught elsewhere - it's not possible to know what specific kinds of exceptions will be thrown and handle them in the usual way.

Reply:
I think if we follow the boy scout rule for this and for other issues then that would be better. As we are in this area of the code lets improve it. I think for now we can catch the specific exceptions that we know about that are going to be thrown and let the exception catch the rest. Ideally, we would catch only the specific ones we know about. There maybe other exceptions that get thrown that we are not aware of but we can't catch everything. Also, on another point it might be worthwhile understanding the root cause of this issue not the symptom. We are catching exceptions here but what is the functionality deeper down. Do we need to address anything in the execute method? Is this expected functionality? Food for thought.

Reply:
The cause of the problem here is the original code not handling the unchecked exception `UnsupportedOperationException` which can be thrown when executing an Operation. This is part of a wider problem of unchecked exception being used instead of checked exceptions.

_In my opinion, this exception shouldn't be thrown or used at all._ As [the official Java guide says,](https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html) unchecked exceptions are for when "code cannot reasonably be expected to recover". In Gaffer we always intend to recover from a situation where an Operation is not supported. As such, this should manifest itself with a checked exception. This can be a child of Gaffer's existing `OperationException`. However, this would be a fairly big change and likely API breaking.

If you think this is a good idea to reduce technical debt, then I can raise this as a separate issue to investigate.

If this improvement were made, then we wouldn't need to catch `Exception` at all or catch anything beyond what is thrown by execute - i.e. there would be no problem and the issue this PR is for wouldn't have existed.

Reply:
Agreed I think we need to revisit this then and to sort out the exceptions that shouldn't be thrown in this way.